### PR TITLE
Baseball update

### PR DESCRIPTION
**Pull request by Couls**
**What does this PR do:**
Adds the badmin bat and pitching machine. Also fixes and tweaks some of the code for baseball bats. 
Badmin bat has no cooldown and deflects any attacks of any size, yes. you can homerun bullets, though it deletes them sadly. It deflects tazer bolts though!
Refer to CL.

The pitching machine has been placed in cargo maints but still needs sprites. it can be unwrenched and moved around. it uses turret sprites as placeholder.

**Images of sprite/map changes (IF APPLICABLE):**
image
ezgif com-video-to-gif

**Changelog:**
:cl:
add: badmin bat and pitching machine.
tweak: bat deflection now deflects in an 180 degree angle from the direction the deflector is facing
fix: fixed an issue with attack animations not showing when batting
fix: fixed an issue with the timer not showing the minutes left before you could deflect again
fix: baseball bats no longer deflect at the speed of light. you can now see deflected objects in flight
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -171,50 +171,64 @@
...
+	icon_state = "baseball_bat_metal"
+	item_state = "baseball_bat_metal"
+
+/obj/item/melee/baseball_bat/badmin/can_deflect()
+	return TRUE // ALWAYS DEFLECT ALWAYS
+
 /obj/item/melee/baseball_bat/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance, damage, attack_type, atom/movable/AM)
 	. = ..()
-	if(!istype(AM, /obj/item) || attack_type != THROWN_PROJECTILE_ATTACK)
+	if(!istype(AM, /obj/item) || attack_type != THROWN_PROJECTILE_ATTACK && !istype(src, /obj/item/melee/baseball_bat/badmin))
```
Comment (Couls) [inline code/game/objects/items/weapons/weaponry.dm]: badmin bat doesn't care for your BULLETS

File: code/game/objects/items/weapons/weaponry.dm [hunk 2]
```diff
@@ -171,50 +171,64 @@
...
+				visible_message("<span class='boldwarning'>[owner] deflects [I] directly back at the thrower! It's a home run!</span>", "<span class='boldwarning'>You deflect the [I] directly back at the thrower! It's a home run!</span>")
 				playsound(get_turf(owner), 'sound/weapons/homerun.ogg', 100, 1)
-				do_attack_animation(I, ATTACK_EFFECT_DISARM)
-				I.throw_at(I.thrownby, 20, 20, owner)
-				deflectmode = FALSE
+				owner.do_attack_animation(I, ATTACK_EFFECT_DISARM)
+				I.throw_at(I.thrownby, 20, 3, owner)
 				if(!istype(I, /obj/item/beach_ball))
 					lastdeflect = world.time + 3000
+				deflectmode = can_deflect()
```
Comment (Couls) [inline code/game/objects/items/weapons/weaponry.dm]: deflectmode stays true for badmin bats

File: code/game/objects/items/weapons/weaponry.dm [hunk 3]
```diff
@@ -171,50 +171,64 @@
...
 				return FALSE
 			else
 				visible_message("<span class='warning'>[owner] swings and deflects [I]!</span>", "<span class='warning'>You swing and deflect the [I]!</span>")
 				playsound(get_turf(owner), 'sound/weapons/baseball_hit.ogg', 50, 1, -1)
-				do_attack_animation(I, ATTACK_EFFECT_DISARM)
-				I.throw_at(get_edge_target_turf(owner, pick(cardinal)), rand(8,10), 14, owner)
-				deflectmode = FALSE
+				owner.do_attack_animation(I, ATTACK_EFFECT_DISARM)
+				var/directionangle = dir2angle(owner.dir)
+				I.throw_at(get_edge_target_turf(owner, angle2dir(rand(directionangle - 180, directionangle + 180))), rand(8,10), 1, owner)
```
Comment (Couls) [inline code/game/objects/items/weapons/weaponry.dm]: should I make this calculation neater?

File: code/game/objects/items/weapons/weaponry.dm [hunk 4]
```diff
@@ -171,50 +171,64 @@
...
+	return TRUE // ALWAYS DEFLECT ALWAYS
+
 /obj/item/melee/baseball_bat/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance, damage, attack_type, atom/movable/AM)
 	. = ..()
-	if(!istype(AM, /obj/item) || attack_type != THROWN_PROJECTILE_ATTACK)
+	if(!istype(AM, /obj/item) || attack_type != THROWN_PROJECTILE_ATTACK && !istype(src, /obj/item/melee/baseball_bat/badmin))
 		return FALSE
 	var/obj/item/I = AM
-	if(I.w_class <= WEIGHT_CLASS_NORMAL || istype(I, /obj/item/beach_ball)) // baseball bat deflecting
+	if(I.w_class <= WEIGHT_CLASS_NORMAL || istype(I, /obj/item/beach_ball) || istype(src, /obj/item/melee/baseball_bat/badmin)) // baseball bat deflecting
```
Comment (Couls) [inline code/game/objects/items/weapons/weaponry.dm]: badmin bat deflects nearly everything

File: code/game/machinery/baseball_machine.dm [hunk 5]
```diff
@@ -0,0 +1,74 @@
+/obj/machinery/baseball_machine
+	name = "baseball pitching machine"
+	desc = "Hey batter batter!"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "turretCover"
+	anchored = TRUE
+	density = TRUE
+	use_power = IDLE_POWER_USE				//this turret uses and requires power
```
Comment (Couls) [inline code/game/machinery/baseball_machine.dm]: copy pasta from turret code

File: code/game/objects/items/weapons/weaponry.dm (hunk 4 above)
Comment (taukausanake) [reply code/game/objects/items/weapons/weaponry.dm]: Can it deflect _mobs_?

Comment (Couls) [reply code/game/objects/items/weapons/weaponry.dm]: god remind me to test that when I add in sprites

File: code/game/machinery/baseball_machine.dm [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	idle_power_usage = 50		//when inactive, this turret takes up constant 50 Equipment power
+	active_power_usage = 300	//when active, this turret takes up constant 300 Equipment power
+	power_channel = EQUIP	//drains power from the EQUIPMENT channel
+	var/baseball_count = 15 // starts with 15 baseballs so we don't just spawn INFINITE amounts of baseballs
+	var/active = FALSE
+
+/obj/machinery/baseball_machine/attackby(obj/item/I, mob/user, params)
+	. = ..()
```
Comment (Fox-McCloud) [inline code/game/machinery/baseball_machine.dm]: This isn't how `attackby` is meant to work

File: code/game/objects/items/weapons/weaponry.dm (hunk 1 above)
Comment (Fox-McCloud) [reply code/game/objects/items/weapons/weaponry.dm]: Did you really.....*REALLY* do an `istype(src)` check.

Bad. And not just bad. *Terrible*

Comment (Citinited) [reply code/game/objects/items/weapons/weaponry.dm]: It just ain't cricket

File: code/game/machinery/baseball_machine.dm [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+	shoot_baseball()
+
+/obj/machinery/baseball_machine/proc/shoot_baseball()
+	if(!baseball_count)
+		atom_say("Out of baseballs. Please insert more baseballs to continue.")
+		active = FALSE
+		update_icon()
+		return FALSE // returns false if ball was not fired
+	var/obj/item/beach_ball/holoball/bball = new /obj/item/beach_ball/holoball/baseball(get_turf(src))
+	bball.throw_at(get_edge_target_turf(src, dir), 20, 1)
```
Comment (Citinited) [inline code/game/machinery/baseball_machine.dm]: Does this play a sound? If not please consider it doing so

Comment (Couls) [reply code/game/machinery/baseball_machine.dm]: Yeah, it should also send a message. I’ll change it

File: code/game/machinery/baseball_machine.dm (hunk 6 above)
Comment (Couls) [reply code/game/machinery/baseball_machine.dm]: I’ll look into how other machines do it

File: code/game/objects/items/weapons/weaponry.dm (hunk 1 above)
Comment (Couls) [reply code/game/objects/items/weapons/weaponry.dm]: AAAA

File: code/game/machinery/baseball_machine.dm [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+/obj/machinery/baseball_machine/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(default_unfasten_wrench(user, I))
+		return
+
+	if(istype(I, /obj/item/beach_ball/holoball/baseball))
+		if(!user.drop_item())
+			return
+		visible_message("<span class='notice'>[src] loads the baseball machine with a baseball.</span>")
+		qdel(I)
```
Comment (Fox-McCloud) [inline code/game/machinery/baseball_machine.dm]: why not store the baseballs in case there's subtypes with unique properties.

Comment (Couls) [reply code/game/machinery/baseball_machine.dm]: Set it up like this originally for testing, I’ll store baseballs inside it instead

File: code/game/machinery/baseball_machine.dm (hunk 6 above)
Comment (Couls) [reply code/game/machinery/baseball_machine.dm]: updated, is this better?

File: code/game/machinery/baseball_machine.dm [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+	idle_power_usage = 50		//when inactive, this turret takes up constant 50 Equipment power
+	active_power_usage = 300	//when active, this turret takes up constant 300 Equipment power
+	power_channel = EQUIP	//drains power from the EQUIPMENT channel
+	var/baseball_count = 15 // starts with 15 baseballs so we don't just spawn INFINITE amounts of baseballs
+	var/active = FALSE
+
+/obj/machinery/baseball_machine/Initialize(mapload)
+	. = ..()
+	for(var/x in 1 to baseball_count)
+		new /obj/item/beach_ball/holoball/baseball(src)
```
Comment (Couls) [inline code/game/machinery/baseball_machine.dm]: fills the machines contents with baseball_count amount of baseballs on initialize

File: code/game/machinery/baseball_machine.dm [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+/obj/machinery/baseball_machine/attackby(obj/item/I, mob/user, params)
+	if(default_unfasten_wrench(user, I))
+		return
+
+	if(istype(I, /obj/item/beach_ball))
```
Comment (Couls) [inline code/game/machinery/baseball_machine.dm]: can load any type of ball now

File: code/game/machinery/baseball_machine.dm [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+	desc = "Hey batter batter!"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "turretCover"
+	anchored = TRUE
+	density = TRUE
+	use_power = IDLE_POWER_USE				//this turret uses and requires power
+	idle_power_usage = 50		//when inactive, this turret takes up constant 50 Equipment power
+	active_power_usage = 300	//when active, this turret takes up constant 300 Equipment power
+	power_channel = EQUIP	//drains power from the EQUIPMENT channel
+	var/baseball_count = 15 // starts with 15 baseballs so we don't just spawn INFINITE amounts of baseballs
```
Comment (Fox-McCloud) [inline code/game/machinery/baseball_machine.dm]: This is gross; much better to add the baseballs to an actual list, then have them moved out of the machine and thrown--that way you're not needlessly creating/destroying objects.

File: code/game/machinery/baseball_machine.dm [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+		new /obj/item/beach_ball/holoball/baseball(src)
+
+/obj/machinery/baseball_machine/attackby(obj/item/I, mob/user, params)
+	if(default_unfasten_wrench(user, I))
+		return
+
+	if(istype(I, /obj/item/beach_ball))
+		if(!user.drop_item())
+			return
+		visible_message("<span class='notice'>[src] loads the baseball machine with a [I].</span>")
```
Comment (Fox-McCloud) [inline code/game/machinery/baseball_machine.dm]: So you can load an unlimited amount of balls into this thing?

File: code/game/machinery/baseball_machine.dm [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+		return
+	shoot_baseball()
+
+/obj/machinery/baseball_machine/proc/shoot_baseball()
+	if(!baseball_count)
+		atom_say("Out of baseballs. Please insert more baseballs to continue.")
+		active = FALSE
+		update_icon()
+		return FALSE // returns false if ball was not fired
+	var/obj/item/beach_ball/bball = locate() in src
```
Comment (Fox-McCloud) [inline code/game/machinery/baseball_machine.dm]: This is so gross.

File: code/game/machinery/baseball_machine.dm [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+	set src in oview(1)
+
+	if(user.incapacitated())
+		return
+	if(!Adjacent(user))
+		return
+	if(!user.IsAdvancedToolUser())
+		to_chat(user, "<span class='warning'>You don't have the dexterity to do that!</span>")
+		return
+	dir = turn(dir, 90)
```
Comment (Fox-McCloud) [inline code/game/machinery/baseball_machine.dm]: use the proper proc, please.

File: code/game/machinery/baseball_machine.dm (hunk 12 above)
Comment (Couls) [reply code/game/machinery/baseball_machine.dm]: If you can get unlimited balls yes, but you’re not given more than 15 to start

File: code/game/machinery/baseball_machine.dm (hunk 14 above)
Comment (Couls) [reply code/game/machinery/baseball_machine.dm]: can you elaborate? looking at the rotate code for other machinery this is how they handled rotation

**Pull request closed without merging**